payments-core/src/main/java/com/stripe/android/payments/core/authentication/UnsupportedAuthenticator.kt

```diff
@@ -0,0 +1,56 @@
+package com.stripe.android.payments.core.authentication
+
+import com.stripe.android.PaymentRelayStarter
+import com.stripe.android.Stripe
+import com.stripe.android.exception.StripeException
+import com.stripe.android.model.StripeIntent
+import com.stripe.android.model.getRequestCode
+import com.stripe.android.networking.ApiRequest
+import com.stripe.android.view.AuthActivityStarterHost
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * [IntentAuthenticator] to return if there is no available authenticators. Informs the correct
+ * dependency to include for that authenticator.
+ */
+@Singleton
+internal class UnsupportedAuthenticator @Inject constructor(
+    private val paymentRelayStarterFactory: (AuthActivityStarterHost) -> PaymentRelayStarter
+) : IntentAuthenticator {
+    override suspend fun authenticate(
+        host: AuthActivityStarterHost,
+        stripeIntent: StripeIntent,
+        threeDs1ReturnUrl: String?,
+        requestOptions: ApiRequest.Options
+    ) {
+        val exception = stripeIntent.nextActionData?.let {
+            val nextActionType = it::class.java
+            StripeException.create(
+                IllegalArgumentException(
+                    "${nextActionType.simpleName} type is not supported, add " +
+                        "${ACTION_DEPENDENCY_MAP[nextActionType]} in build.gradle to support it"
```

Comment:
Is it possible that there isn't an entry in `ACTION_DEPENDENCY_MAP` for `nextActionType`?

Reply:
not really, this is not a sink class where unsupported types are matched to by default, `NoOpAuthenticator` is used for that. We will manually match a type to `UnsupportedAuthenticator` when reflection fails, so the items routed here are all expected types.